src/repr/src/adt/datetime.rs

```diff
@@ -1199,9 +1196,17 @@ fn fill_pdt_date(
     // Check for one number that represents YYYYMMDDD.
     match actual.front() {
         Some(Num(mut val, digits)) if 6 <= *digits && *digits <= 8 => {
-            pdt.day = Some(DateTimeFieldValue::new(val % 100, 0));
+            pdt.day = Some(DateTimeFieldValue::new(
+                i64::try_from(val % 100)
+                    .expect("modulo operation between u64 and constant 100 should fit signed 64-bit integer"),
+                0,
+            ));
             val /= 100;
-            pdt.month = Some(DateTimeFieldValue::new(val % 100, 0));
+            pdt.month = Some(DateTimeFieldValue::new(
+                i64::try_from(val % 100)
+                .expect("modulo operation between u64 and constant 100 should fit signed 64-bit integer"),
```

Comment:
Strange that this line isn't indented while the same one above is. I'd make that consistent.

Also, it might improve readability a bit by splitting these into two statements, to reduce the nesting:
```rust
let unit = i64::try_from(val % 100)
    .expect("modulo operation between u64 and constant 100 should fit signed 64-bit integer");
pdt.day = Some(DateTimeFieldValue::new(unit, 0));
```